src/marqo/tensor_search/tensor_search.py

```diff
@@ -1657,3 +1663,118 @@ def get_cuda_info() -> dict:
+            #     )
+            #     return chunks_to_append, document_is_valid, unsuccessful_docs, total_vectorise_time
+
+
+            try:
+                if infer_if_image is False:
+                    raise errors.InvalidArgError(
+                        f"The setting `treat_urls_and_pointers_as_images` is `{infer_if_image}` for a multimodal_tensor_combination field."
+                        f"This is not supported. Please change the setting `treat_urls_and_pointers_as_images` as `True` for multimodal_tensor_combination fields. "
+                    )
+            except errors.InvalidArgError as e:
+                document_is_valid = False
+                unsuccessful_docs.append(
+                    (i, {'_id': doc_id, 'error': e.message,
+                         'status': int(errors.InvalidArgError.status_code),
+                         'code': errors.InvalidArgError.code})
+                )
+                return chunks_to_append, document_is_valid, unsuccessful_docs, total_vectorise_time
+            try:
+                start_time = timer()
+                vector_chunks = s2_inference.vectorise(
+                    model_name=index_info.model_name,
+                    model_properties=_get_model_properties(index_info), content=content_chunks,
+                    device=selected_device, normalize_embeddings=normalize_embeddings,
+                    infer=infer_if_image)
+
+                end_time = timer()
+                single_vectorise_call = end_time - start_time
+                total_vectorise_time += single_vectorise_call
+                logger.debug(f"(4) TIME for single vectorise call: {(single_vectorise_call):.3f}s.")
+            except (s2_inference_errors.UnknownModelError,
+                    s2_inference_errors.InvalidModelPropertiesError,
+                    s2_inference_errors.ModelLoadError) as model_error:
+                raise errors.BadRequestError(
+                    message=f'Problem vectorising query. Reason: {str(model_error)}',
+                    link="https://marqo.pages.dev/latest/Models-Reference/dense_retrieval/"
+                )
+            except s2_inference_errors.S2InferenceError:
+                document_is_valid = False
+                image_err = errors.InvalidArgError(message=f'Could not process given image: {field_content}')
+                unsuccessful_docs.append(
+                    (i, {'_id': doc_id, 'error': image_err.message, 'status': int(image_err.status_code),
+                         'code': image_err.code})
+                )
+                return chunks_to_append, document_is_valid, unsuccessful_docs, total_vectorise_time
+
+            field_vector_weight.append((sub_content_para["weight"], vector_chunks))
+
+    vector_chunk = np.squeeze(
+                np.sum([np.array(vector) * weight for weight, vector in field_vector_weight], axis=0) / np.sum(
+            [weight for weight, vector in field_vector_weight])).tolist()
+
```

Comment:
Because we only return one chunk we could just return a chunk, and then append it in the calling function.

Reply:
yes, i am returning one chunk now, like `chunks_to_append`.